### PR TITLE
Edit Partnerships section in the Project View page

### PR DESCRIPTION
**Pull request by sdevalapurkar**
<details open>
  <summary><a href="https://quartech.atlassian.net/browse/BHBC-952" title="BHBC-952" target="_blank">BHBC-952</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>Edit Partnerships section in the Project View page</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Story
        Story
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Progress</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td>-</td>
    </tr>
  </table>
</details>

---

# Overview

## This PR contains the following changes

- Edit Partnerships section in the Project View page

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)

## How Has This Been Tested?

- Locally

### CODE REVIEW

File: app/src/features/projects/view/components/Partnerships.test.tsx [hunk 1]
```diff
@@ -2,28 +2,29 @@ import { render, waitFor, fireEvent, cleanup } from '@testing-library/react';
...
-
-const history = createMemoryHistory();
+import { UPDATE_GET_ENTITIES } from 'interfaces/useProjectApi.interface';
 
 jest.mock('../../../../hooks/useBioHubApi');
 const mockUseBiohubApi = {
   project: {
-    getProjectForUpdate: jest.fn<Promise<object>, []>()
+    getProjectForUpdate: jest.fn<Promise<object>, []>(),
+    updateProject: jest.fn()
```
Comment (anissa-agahchen) [inline app/src/features/projects/view/components/Partnerships.test.tsx:13]: what is the updateProject function for?  I haven't seen it used in other tests, I don't think.

Comment (sdevalapurkar) [reply app/src/features/projects/view/components/Partnerships.test.tsx:13]: its for the actual edit (mocking it)

Comment (sdevalapurkar) [reply app/src/features/projects/view/components/Partnerships.test.tsx:13]: it is in `species.test.tsx`, `projectcoordinator.test.tsx`, `iucnclassification.test.tsx` etc

Comment (anissa-agahchen) [reply app/src/features/projects/view/components/Partnerships.test.tsx:13]: ah ok

**Review by anissa-agahchen — COMMENTED** [review]
Would you be able to add a test for the error dialog?  That's the main reason the codecov went down ... and then we can duplicate it for the other components

File: app/src/features/projects/view/components/Partnerships.tsx [hunk 2]
```diff
@@ -57,29 +58,64 @@ const Partnerships: React.FC<IPartnershipsProps> = (props) => {
...
-    setOpenEditDialog(false);
-    history.push(`/projects/${id}/details`);
+  const handleDialogEditSave = async (values: IProjectPartnershipsForm) => {
+    const projectData = { partnerships: values };
+
+    try {
+      await biohubApi.project.updateProject(id, projectData);
+    } catch (error) {
+      const apiError = new APIError(error);
+      showErrorDialog({ dialogText: apiError.message, open: true });
```
Comment (anissa-agahchen) [inline app/src/features/projects/view/components/Partnerships.tsx:96]: I was referring to a having a test for when we show the error dialog would be good.  If it's too hard, let's not worry about it for now.

**Review by anissa-agahchen — APPROVED** [review]

File: app/src/features/projects/view/components/Partnerships.tsx (hunk 2 above)
Comment (sdevalapurkar) [reply app/src/features/projects/view/components/Partnerships.tsx:96]: will look at it in subsequent PR

**Pull request merged**